### ISSUE TITLE
Inconsistent payment methods display behavior

### ISSUE THREAD

**Issue by alepore** [spree/spree#6421]
TL;DR I think `PaymentMethod.available` is broken and the respective tests are wrong.

While working on https://github.com/spree/spree/pull/6420 i found an inconsistent behavior on scoping payment methods by `visible_on`:
* objects with `display_on` nil or blank are treated as visibile
* `PaymentMethod.available` and `PaymentMethod.available(:both)` does not return objects with display_on == "front_end" or "back_end"
* `PaymentMethod.available(:frontend)` does not return objects with display_on == "both"
* tests are not well written and allow the above

I can fix this but someone should tell me what the actual expected behavior is.

I would expect:
* `display_on` nil or blank = do not display (or just don't allow this and make default = backend)
* `PaymentMethod.available` should return front_end, back_end, and both
* `PaymentMethod.available(:frontend)` should return objects with display_on == "frontend" and "both"

Thanks!!

**Comment by JDutil** [comment]
@alepore the current display_on being blank means display on both.  It'd be fine to clean this up, but we need to at least migrate peoples data so blank becomes 'both'.

**Comment by nishant-cyro** [comment]
@priyank-gupta @Mafi88 @damianlegawiec - We can close this one as current master branch follows the behaviour specified in this PR. Please refer https://github.com/spree/spree/commit/d1574446.

**Comment by Mafi88** [comment]
@nishant-cyro :+1:
